test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommand_PackagesLockFileTests.cs

```diff
@@ -0,0 +1,951 @@
+                    .WithTargetFrameworks(new string[] { targetFramework.GetShortFolderName() })
+                    .Build();
+                allPackageSpecs.Add(leafProject1PackageSpec);
+
+                // Add the dependency to all frameworks
+                PackageSpecOperations.AddOrUpdateDependency(rootPackageSpec, packageA.Identity, rootPackageSpec.TargetFrameworks.Select(e => e.FrameworkName));
+                PackageSpecOperations.AddOrUpdateDependency(intermediatePackageSpec, packageB.Identity, intermediatePackageSpec.TargetFrameworks.Select(e => e.FrameworkName));
+                PackageSpecOperations.AddOrUpdateDependency(leafProject1PackageSpec, packageC.Identity, leafProject1PackageSpec.TargetFrameworks.Select(e => e.FrameworkName));
+                PackageSpecOperationsUtility.AddProjectReference(rootPackageSpec, intermediatePackageSpec, targetFramework);
+                PackageSpecOperationsUtility.AddProjectReference(intermediatePackageSpec, leafProject1PackageSpec, targetFramework);
+
+                // Preconditions.
+                var result = await new RestoreCommand(ProjectTestHelpers.CreateRestoreRequest(rootPackageSpec, allPackageSpecs, pathContext, logger)).ExecuteAsync();
+                await result.CommitAsync(logger, CancellationToken.None);
+                result.Success.Should().BeTrue();
+
+                // Enable locked mode, remove a dependency and clear the logger;
+                rootPackageSpec.RestoreMetadata.RestoreLockProperties = new RestoreLockProperties(
+                    restorePackagesWithLockFile: "true",
+                    rootPackageSpec.RestoreMetadata.RestoreLockProperties.NuGetLockFilePath,
+                    restoreLockedMode: true);
+                logger.Clear();
+
+                projectName = "LeaftProject2";
+                projectDirectory = Path.Combine(pathContext.SolutionRoot, projectName);
+                var leafProject2PackageSpec = PackageReferenceSpecBuilder.Create(projectName, projectDirectory)
+                    .WithTargetFrameworks(new string[] { targetFramework.GetShortFolderName() })
+                    .Build();
+                PackageSpecOperationsUtility.AddProjectReference(intermediatePackageSpec, leafProject2PackageSpec, targetFramework);
+                allPackageSpecs.Add(leafProject2PackageSpec);
+
+                // Act.
+                result = await new RestoreCommand(ProjectTestHelpers.CreateRestoreRequest(rootPackageSpec, allPackageSpecs, pathContext, logger)).ExecuteAsync();
+
+                // Assert.
+                result.Success.Should().BeFalse();
+                logger.ErrorMessages.Count.Should().Be(1);
+                logger.ErrorMessages.Single().Should().Contain("NU1004");
+                logger.ErrorMessages.Single().Should().Contain($"The project reference intermediateproject has changed. Current dependencies count: 3, lock file's dependencies count: 2.");
```

Comment:
Should we add the list of dependencies instead like we do in the other messages?

Reply:
Updated the message to list the dependencies instead.